src/wazuh_modules/syscollector/sysInfoWin.cpp

```diff
@@ -8,18 +8,280 @@
  * License (version 2) as published by the FSF - Free Software
  * Foundation.
  */
-#include "sysInfo.hpp"
+
+#include <winsock2.h>
+#include <windows.h>
+#include <psapi.h>
+#include <tlhelp32.h>
 #include <memory>
 #include <list>
 #include <system_error>
+#include <versionhelpers.h>
+#include "sysinfoapi.h"
+#include "sysInfo.hpp"
 #include "cmdHelper.h"
 #include "stringHelper.h"
 #include "registryHelper.h"
-#include "sysinfoapi.h"
-#include <versionhelpers.h>
+#include "defs.h"
+#include "debug_op.h"
 
 constexpr auto BASEBOARD_INFORMATION_TYPE{2};
 constexpr auto CENTRAL_PROCESSOR_REGISTRY{"HARDWARE\\DESCRIPTION\\System\\CentralProcessor\\0"};
+constexpr auto SYSTEM_IDLE_PROCESS_NAME{"System Idle Process"};
+constexpr auto SYSTEM_PROCESS_NAME{"System"};
+
+struct CharDeleter
+{
+    void operator()(char* buffer)
+    {
+        free(buffer);
+    }
+};
+
+static bool isVistaOrLater()
+{
+    static const bool ret
+    {
+        IsWindowsVistaOrGreater()
+    };
+    return ret;
+}
+
+class SysInfoProcess final
+{
+public:
+    SysInfoProcess(const DWORD pId)
+        : m_pId(pId),
+          m_kernelModeTime{},
+          m_userModeTime{}
+    {
+        m_spProcess = OpenProcess(PROCESS_QUERY_INFORMATION | PROCESS_VM_READ, FALSE, m_pId);
+        if (m_spProcess)
+        {
+            setProcessTimes();
+            setProcessMemInfo();
+        }
+        else
+        {
+            throw std::system_error
+            {
+                static_cast<int>(GetLastError()),
+                std::system_category(),
+                "Unable to open process with PID %lu" + m_pId
+            };
+        }
+    }
+
+    ~SysInfoProcess()
+    {
+        CloseHandle(m_spProcess);
+    }
+
+    std::string cmd()
+    {
+        std::string ret { "unknown" };
+        char readBuffer[OS_MAXSTR] = {};
+        //std::unique_ptr<char*, CharDeleter> path { nullptr };
+        char* path { nullptr };
+        // Get full Windows kernel path for the process
+        if (GetProcessImageFileName(m_spProcess, readBuffer, OS_MAXSTR))
+        {
+            // Convert Windows kernel path to a valid Win32 filepath
+            // E.g.: "\Device\HarddiskVolume1\Windows\system32\notepad.exe" -> "C:\Windows\system32\notepad.exe"
+            // This requires hotfix KB931305 in order to work under XP/Server 2003, so the conversion will be skipped if we're not running under Vista or greater
+            if (!isVistaOrLater() || !ntPath2Win32Path(readBuffer, &path))
+            {
+                // If there were any errors, the readBuffer array will remain intact
+                // In that case, let's just use the Windows kernel path. It's better than nothing
+                ret = path;
```

Comment:
if !isVistaOrLater retunrs true ret will be set to nullptr because path is nullptr. Is that the expected behavior?

Reply:
Will do.